From cc39f32ad6aca616b3390125474b7bf40a8ae602 Mon Sep 17 00:00:00 2001
From: "Subhransu S. Prusty" <subhransu.s.prusty@intel.com>
Date: Fri, 2 May 2014 23:52:25 +0530
Subject: [PATCH 31/39] ASoC: Intel: DFW: Remove const from sst_pcm_format
 structure

Issue: GMINL-3680

Remove const from sst_pcm_format struct else it will
create issue when assigning value from DFW binary

Change-Id: I7b80250a2214a21ba846030ebb0914c9f18600be
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
---
 sound/soc/intel/platform-libs/controls_v2_dpcm.c | 4 ++--
 sound/soc/intel/platform-libs/sst_widgets.h      | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/intel/platform-libs/controls_v2_dpcm.c b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
index 2c2e199..069b3cf 100644
--- a/sound/soc/intel/platform-libs/controls_v2_dpcm.c
+++ b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
@@ -1344,13 +1344,13 @@ static const struct snd_kcontrol_new sst_bt_fm_mux =
 	SST_SSP_MUX_CTL("ssp1_out", 0, SST_MUX_REG, SST_BT_FM_MUX_SHIFT, sst_bt_fm_texts,
 			sst_mux_get, sst_mux_put);
 
-static const struct sst_pcm_format aware_stream_fmt = {
+static struct sst_pcm_format aware_stream_fmt = {
 	.sample_bits = 24,
 	.rate_min = 8000,
 	.channels_max = 1,
 };
 
-static const struct sst_pcm_format vad_stream_fmt = {
+static struct sst_pcm_format vad_stream_fmt = {
 	.sample_bits = 16,
 	.rate_min = 16000,
 	.channels_max = 1,
diff --git a/sound/soc/intel/platform-libs/sst_widgets.h b/sound/soc/intel/platform-libs/sst_widgets.h
index b0b4356..2961d3d 100644
--- a/sound/soc/intel/platform-libs/sst_widgets.h
+++ b/sound/soc/intel/platform-libs/sst_widgets.h
@@ -45,7 +45,7 @@ struct sst_ids {
 	struct snd_soc_dapm_widget *parent_w;
 	struct list_head algo_list;
 	struct list_head gain_list;
-	const struct sst_pcm_format *pcm_fmt;
+	struct sst_pcm_format *pcm_fmt;
 };
 
 
-- 
1.9.1

